public/app/core/navigation/types.ts

```diff
@@ -13,7 +13,7 @@ export type GrafanaRouteComponent<T extends {} = any> = React.ComponentType<Graf
 export interface RouteDescriptor {
   path: string;
   component: GrafanaRouteComponent;
-  roles?: () => string[];
+  roles?: () => string[]; // TODO(aarongodin): understand what this means
```

Comment:
yeah, that's a bit tricky. We use it for frontend router to protect routes with specific roles. But this `roles` property only for basic roles, we handle RBAC roles [here](https://github.com/grafana/grafana/blob/eef07aedc89e4891b844227b13f69c6946120f2b/public/app/features/plugins/admin/routes.tsx#L43). I would like to simplify this a bit in a future.

Reply:
Here we check the roles https://github.com/grafana/grafana/blob/eef07aedc89e4891b844227b13f69c6946120f2b/public/app/AppWrapper.tsx#L68